### PR TITLE
Add NNKolmogorov and NNParamKolmogorov

### PR DESCRIPTION
**Pull request by ashutosh-b-b**
## Checklist

- [x] Appropriate tests were added
- [x] Any code changes were done in a way that does not break public API
- [x] All documentation related to code changes were updated
- [x] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [x] Any new documentation only uses public API

### CODE REVIEW

**Comment by ashutosh-b-b** [comment]
Needs #81 to go in first.

File: docs/src/NNParamKolmogorov.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+```@autodocs
+Modules = [HighDimPDE]
+Pages   = ["NNParamKolmogorov.jl"]
+```
+
+`NNParamKolmogorov` obtains a terminal solution for parametric families of Backward Kolmogorov Equations of the form:
+```math
+\partial_t u(t,x) = \mu(t, x, γ_mu) \nabla_x u(t,x) + \frac{1}{2} \sigma^2(t, x, γ_sigma) \Delta_x u(t,x)
+```
+with initial condition given by `g(x, γ_phi)`
```
Comment (ChrisRackauckas) [inline docs/src/NNParamKolmogorov.md:15]: terminal condition? I'm not sure this really matches??

Comment (ashutosh-b-b) [reply docs/src/NNParamKolmogorov.md:15]: image

Incase of forward eqns it is the initial condition.

Comment (ChrisRackauckas) [reply docs/src/NNParamKolmogorov.md:15]: Yes in forward equations it's an initial condition. But for backwards we have a terminal condition. How are the two separate cases being handled?

Comment (ashutosh-b-b) [reply docs/src/NNParamKolmogorov.md:15]: Isn't in this case the conversion as simple as `v(t,x) = u(T - t, x)` for changing from fwd to backward. I found this: 
image

I got this from here : https://jberner.info/data/Thesis_Berner.pdf

**Review by ChrisRackauckas — APPROVED** [review]

**Pull request merged**